xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-configuration/xwiki-platform-rendering-configuration-default/src/test/java/org/xwiki/rendering/internal/configuration/DefaultExtendedRenderingConfigurationTest.java

```diff
@@ -48,77 +54,79 @@
  * @version $Id$
  * @since 8.2M1
  */
+@ComponentTest
 public class DefaultExtendedRenderingConfigurationTest
 {
-    @Rule
-    public MockitoComponentMockingRule<DefaultExtendedRenderingConfiguration> mocker =
-        new MockitoComponentMockingRule<>(DefaultExtendedRenderingConfiguration.class);
+    @InjectComponentManager
+    private MockitoComponentManager componentManager;
 
-    @Before
-    public void configure() throws Exception
+    @InjectMockComponents
+    private DefaultExtendedRenderingConfiguration defaultExtendedRenderingConfiguration;
+
+    @BeforeComponent
+    public void setup() throws Exception
     {
-        Provider<ComponentManager> contextComponentManagerProvider = this.mocker.registerMockComponent(
-            new DefaultParameterizedType(null, Provider.class, ComponentManager.class), "context");
-        when(contextComponentManagerProvider.get()).thenReturn(this.mocker);
+        Provider<ComponentManager> contextComponentManagerProvider = this.componentManager.registerMockComponent(
+                new DefaultParameterizedType(null, Provider.class, ComponentManager.class), "context");
+        when(contextComponentManagerProvider.get()).thenReturn(this.componentManager);
     }
 
     @Test
     public void getImageWidthLimit() throws Exception
     {
-        ConfigurationSource source = this.mocker.getInstance(ConfigurationSource.class);
+        ConfigurationSource source = componentManager.getInstance(ConfigurationSource.class);
```

Comment:
It's indeed an exact conversion of the test case to Junit 5 with the same logic but I feel that it would be simpler to understand with

    @MockComponent
    private ConfigurationSource source;

    @MockComponent
    @Named("rendering")
    private ConfigurationSource renderingSource;

    ....

    when(this.source.getProperty("rendering.imageWidthLimit", -1)).thenReturn(100);

    ....

    when(this.renderingSource.getProperty("disabledSyntaxes")).thenReturn(null);

instead of going trough the component manager.

Reply:
Hm indeed! Actually I just refactor the test syntactically here, but I need to fix the pseudo-tested methods: so I'll certainly fix that one with the change you propose.